tools/snp.sh

```diff
@@ -904,13 +914,168 @@ attest_guest() {
     || { >&2 echo -e "FAIL: measurements do not match"; return 1; }
 }
 
+# Additional Functions Added here
+rhel_subscription_mgr_set_login(){
+  echo "Enter RedHat subscription Manager credentials"
+  read -p "Username: " RHEL_SUBS_MGR_USER
+  read -sp "Password: " RHEL_SUBS_MGR_PASS
+}
+
+set_cloud_init_url_based_on_linux_distribution(){
+  case ${ID,,} in
+  ubuntu)
+    CLOUD_INIT_IMAGE_URL="https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img"
+    ;;
+  rhel)
+    # Can't Initialize CLOUD_INIT_IMAGE_URL for redhat due to redhat subscription requirement
+    echo "Download Red Hat Enterprise Linux 9.2 KVM Guest Image from RedHat Login" 
+    ;;
+  esac
+}
+
+check_if_dependencies_installed(){
+  local dependencies_installed_file="${WORKING_DIR}/dependencies_already_installed"
+  source "${HOME}/.cargo/env" 2>/dev/null || true
+
+  if [ -f "${dependencies_installed_file}" ]; then
+    echo -e "Dependencies previously installed"
+    dependencies_installed=1
+  fi
+}
+
+install_common_dependencies(){
+  # pip issue on 20.04 - some openssl bug
+  #sudo rm -f "/usr/lib/python3/dist-packages/OpenSSL/crypto.py"
+  pip install sev-snp-measure
+
+  # Rust is required to build snpguest
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
```

Comment:
We should not assume that the program is installed. Let's make sure we check before we call it.